selftests/unit/test_parameters.py

```diff
@@ -21,3 +21,35 @@ def test_greedy_path_to_re(self):
                          'foo/')
         self.assertEqual(params._greedy_path_to_re('/foo/*').pattern,
                          '/foo/')
+
+    def test_same_origin_of_different_nodes(self):
+        # ideally we have one tree, therefor shared key
+        # have identical origin (id of the origin env)
+        foo = tree.TreeNode().get_node("/foo", True)
+        root = foo.parent
+        root.value = {'timeout': 1}
+        bar = root.get_node("/bar", True)
+        params1 = parameters.AvocadoParams([foo, bar], '/')
+        self.assertEqual(params1.get('timeout'), 1)
+        self.assertEqual(params1.get('timeout', '/foo/'), 1)
+        self.assertEqual(params1.get('timeout', '/bar/'), 1)
+        # Sometimes we get multiple trees, but if they clame the origin
+        # is of the same path, let's trust it (even when the value is
+        # differs)
+        # note: This is artificial example which should not happen in
+        # in production. Anyway in json-loader we do create only
```

Comment:
Another nipick: maybe be more clear about json-*variants*-loader

Reply:
could be